controllers/component_controller_conditions.go

```diff
@@ -50,7 +59,7 @@ func (r *ComponentReconciler) SetCreateConditionAndUpdateCR(ctx context.Context,
 	}
 	meta.SetStatusCondition(&component.Status.Conditions, condition)
 
-	err := r.Client.Status().Update(ctx, component)
+	err = r.Client.Status().Patch(ctx, component, patch)
```

Comment:
I think we need to pass in the address of the object `component` according to the [api docs](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.14.4/pkg/client/interfaces.go#LL75C7-L75C7).  This is also being done in the ASEB [patches](https://github.com/redhat-appstudio/application-service/blob/main/controllers/applicationsnapshotenvironmentbinding_controller_conditions.go#L61)

Reply:
the `component` is already passed in as `*appstudiov1alpha1.Component` type